packages/radio/src/index.tsx

```diff
@@ -0,0 +1,378 @@
+ */
+export type RadioGroupProps = {
+  /**
+   * Like form inputs, a radio group's state can be controlled by the owner.
+   * Make sure to include an `onChange` as well, or else the radio group will
+   * not be interactive.
+   *
+   * @see Docs https://reacttraining.com/reach-ui/radio#radiogroup-value
+   */
+  value?: RadioValue | null;
+  /**
+   * The default selected value for an uncontrolled radio group component.
+   *
+   * @see Docs https://reacttraining.com/reach-ui/radio#radiogroup-defaultvalue
+   */
+  defaultValue?: RadioValue;
+  /**
+   * Whether or not all radio buttons in a group should be disabled.
+   *
+   * @see Docs https://reacttraining.com/reach-ui/radio#radiogroup-disabled
+   */
+  disabled?: boolean;
+  /**
+   * The `name` attribute passed to a hidden form input.
+   *
+   * @see Docs https://reacttraining.com/reach-ui/radio#radiogroup-name
+   */
+  name?: React.InputHTMLAttributes<HTMLInputElement>["name"];
+  /**
+   * Calls back with the value whenever the user changes the active radio
+   * button.
+   *
+   * @see Docs https://reacttraining.com/reach-ui/tabs#tabs-props
+   */
+  onChange?: RadioChangeHandler;
+};
+
+////////////////////////////////////////////////////////////////////////////////
+
+/**
+ * Radio
+ *
+ * @see Docs https://reacttraining.com/reach-ui/radio#radio-1
+ */
+export const Radio = forwardRefWithAs<RadioProps, "span">(function Radio(
+  {
+    as: Comp = "span",
+    children,
+    onBlur,
+    onClick,
+    onFocus,
+    onKeyDown,
+    disabled: disabledProp,
+    readOnly,
+    value,
+    ...props
+  },
+  forwardedRef
+) {
+  let {
+    disabled: groupDisabled,
+    isRTL,
+    name,
+    onChange,
+    send,
+    value: groupValue,
+  } = useRadioGroupContext();
+  let { descendants: radioOptions } = useContext(RadioDescendantsContext);
+
+  let disabled = groupDisabled || !!disabledProp;
+
+  let ownRef = useRef<HTMLElement | null>(null);
+  let ref = useForkedRef(forwardedRef, ownRef);
+
+  let inputRef = useRef<HTMLInputElement | null>(null);
+
+  let focusableOptions = useMemo(
+    () => radioOptions.filter(option => option.element && !option.disabled),
+    [radioOptions]
+  );
+
+  let isSelected = value === groupValue;
+
+  // A radio button is only tabbable if it is either selected OR if no buttons
+  // are selected and it the first focusable option in our list.
+  let isTabbable =
+    isSelected ||
+    (groupValue === null &&
+      focusableOptions[0] &&
+      focusableOptions[0].element === ownRef.current);
+
+  let _id = useId(props.id);
+  let id = props.id ?? makeId("radio", value, _id);
+  let labelId = makeId("label", id);
+
+  function handleClick(event: React.MouseEvent) {
+    send({
+      type: RadioGroupEvents.Select,
+      callback: onChange,
+      disabled,
+      domEvent: event.nativeEvent,
+      node: ownRef.current,
+      value,
+    });
+  }
+
+  function handleKeyDown(event: React.KeyboardEvent) {
```

Comment:
I know it didn't exist at the time of this PR, but what do you think about using the [`useDescendantKeydown`](https://github.com/reach/reach-ui/blob/52d721cd643d1a6d24c7253ab24e114be262d092/packages/descendants/src/index.tsx#L221) abstraction here?